api/common/network.go

```diff
@@ -35,9 +35,13 @@ type NetworkConfigSource interface {
 	// default route on the machine. If there is no default route (known),
 	// then zero values are returned.
 	DefaultRoute() (net.IP, string, error)
+
+	Origin() corenetwork.Origin
```

Comment:
To propagate this through to the construction of `params.NetworkConfig` we can define a custom Origin. If it does turn out that container and container_initialisation are `OriginMachine` then we can probably revert this part and hard code it on the construction of the `params.NetworkConfig`.